$OpenBSD: patch-gtk_gtkfilechooserwidget_c,v 1.5 2020/05/19 12:00:40 ajacoutot Exp $

unveil(2): check if the XDG directories exist before showing them on the sidebar

Index: gtk/gtkfilechooserwidget.c
--- gtk/gtkfilechooserwidget.c.orig
+++ gtk/gtkfilechooserwidget.c
@@ -7176,7 +7176,8 @@ desktop_folder_handler (GtkFileChooserWidget *impl)
    * See http://freedesktop.org/wiki/Software/xdg-user-dirs
    */
   name = g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP);
-  if (!g_strcmp0 (name, g_get_home_dir ()))
+  if (!g_strcmp0 (name, g_get_home_dir ()) ||
+      !g_file_test (name, G_FILE_TEST_IS_DIR))
     return;
 
   file = g_file_new_for_path (name);
